pulp_2to3_migrate/app/serializers.py

```diff
@@ -29,12 +35,44 @@ def validate(self, data):
         """
         Validate that the Serializer contains valid data.
 
-        TODO:
-        Validate JSON structure of migration_plan.
-        Check validity of the JSON content:
-         - migration for requested plugins is supported
-
+        Validates JSON structure of migration_plan.
+        Checks pulp2 and pulp3 plugins are installed.
         """
+        schema = json.loads(SCHEMA)
+        validator = Draft7Validator(schema)
+        loaded_plan = json.loads(data['plan'])
+        err = []
+        for error in sorted(validator.iter_errors(loaded_plan), key=str):
+            err.append(error.message)
+        if err:
+            raise serializers.ValidationError(
+                    _("Provided Migration Plan format is invalid:'{}'".format(err))
+            )
+        plugins_to_migrate = set()
+        for plugin_type in loaded_plan['plugins']:
+            plugins_to_migrate.add(plugin_type['type'])
+        if len(loaded_plan['plugins']) != len(plugins_to_migrate):
+            raise serializers.ValidationError(
+                _("Provided Migration Plan contains same plugin type specified more that once.")
+            )
+        # MongoDB connection initialization
+        connection.initialize()
+        db = connection.get_database()
+        for plugin in plugins_to_migrate:
+            plugin_type = 'pulp_{plugin}'.format(plugin=plugin)
+            if plugin_type not in INSTALLED_PULP_PLUGINS:
+                raise serializers.ValidationError(
+                    _("Plugin {} is not installed in pulp3.".format(plugin))
+                )
+            try:
+                # this will not work for plugins that have more that one content
+                # maybe provide mapping plugin --> pulp2 content types?
+                plugin_type = 'units_{plugin}'.format(plugin=plugin)
+                db.command("collstats", plugin_type)
```

Comment:
this is the only way i found how to check whether a collection exists, suggestions welcome